popout.js

```diff
@@ -1,241 +1,355 @@
-							// but we don't want to render them
-							ui.chat = new ChatLog({tabName: "chat"})
-							ui.combat = new CombatTracker({tabName: "combat"})
-							ui.scenes = new SceneDirectory({tabName: "scenes"})
-							ui.actors = new ActorDirectory({tabName: "actors"})
-							ui.items = new ItemDirectory({tabName: "items"})
-							ui.journal = new JournalDirectory({tabName: "journal"})
-							ui.tables = new RollTableDirectory({tabName: "tables"})
-							ui.playlists = new PlaylistDirectory({tabName: "playlists"})
-							ui.compendium = new CompendiumDirectory({tabName: "compendium"})
-							ui.settings = new Settings({tabName: "settings"})
-					}
-							
-				KeyboardManager.prototype._onEscape = function(event, up, modifiers) {
-					if ( up || modifiers.hasFocus ) return;
-
-					// Case 1 - dismiss an open context menu
-					if ( ui.context && ui.context.menu.length ) ui.context.close();
-
-					// Case 2 - close open UI windows
-					else if ( Object.keys(ui.windows).length ) {
-						Object.values(ui.windows).filter(w => w.id !== ${sheetGetter}.id).forEach(app => app.close());
-					}
-
-					// Flag the keydown workflow as handled
-					this._handled.add(event.keyCode);
-				}
-				// Add delay before rendering in case some things aren't done initializing, like sheet templates
-				// which get loaded asynchronously.
-				Hooks.on('ready', () => setTimeout(() => PopoutModule.${renderer}(${sheetGetter}), ${timeout}));
-		  	window.dispatchEvent(new Event('load'))
-		  </script>`))
-		// Open new window and write the new html document into it
-		// We need to open it to the same url because some images use relative paths
-		let windowFeatures = undefined;
-		if (game.settings.get("popout", "useWindows")) {
-            const innerWidth = sheet.element.innerWidth() + padding * 2;
-            const innerHeight = sheet.element.innerHeight() + padding * 2;
-            const position = sheet.element.position();
+    constructor() {
+        this.poppedOut = new Map();
+        this.TIMEOUT_INTERVAL = 25 // ms
+        this.MAX_TIMEOUT = 250; // ms
+        // Random id to prevent collision with other modules;
+        this.ID = [...Array(24)].map(() => Math.floor(Math.random() * 16).toString(16)).join("");
+    }
+
+    log(msg, ...args) {
+        if (game && game.settings.get("popout", "verboseLogs")) {
+            const color = "background: #6699ff; color: #000; font-size: larger;";
+            console.debug(`%c PopoutModule: ${msg}`, color, ...args);
+        }
+    }
+
+    async init() {
+        game.settings.register("popout", "useWindows", {
+            name: "Pop sheets out into windows",
+            hint: "Force the popped out sheet to be a window with minimal decorations. Otherwise uses your browser's default (a new tab most likely)",
+            scope: "client",
+            config: true,
+            default: false,
+            type: Boolean
+        });
+
+        game.settings.register("popout", "verboseLogs", {

```

Comment:
Correct, and I would say there are probably subtle bugs if you have two copies of an application altering state at the same time. In theory it should work, but I suspect because you can't log into an account multiple times, and therefore in the base implementation you shouldn't be able to have 2 copies of the same app open, there is a lot of implicit assumptions backed into both the core game and external modules about where updates to an applications state came come from.